src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -0,0 +1,219 @@
+ * classification algorithms.
+ *
+ * To construct a KFoldCV object you need to pass the k parameter and arguments
+ * that specify data. For the latter see the CVBase constructors as a reference
+ * - the CVBase constructors take exactly the same arguments as ones that are
+ * supposed to be passed after the k parameter in the KFoldCV constructor.
+ *
+ * For example, you can run 10-fold cross-validation for SoftmaxRegression in
+ * the following way.
+ *
+ * @code
+ * // 100-point 5-dimensional random dataset.
+ * arma::mat data = arma::randu<arma::mat>(5, 100);
+ * // Random labels in the [0, 4] interval.
+ * arma::Row<size_t> labels =
+ *     arma::randi<arma::Row<size_t>>(100, arma::distr_param(0, 4));
+ * size_t numClasses = 5;
+ *
+ * KFoldCV<SoftmaxRegression<>, Accuracy> cv(10, data, labels, numClasses);
+ *
+ * double lambda = 0.1;
+ * double softmaxAccuracy = cv.Evaluate(lambda);
+ * @endcode
+ *
+ * @tparam MLAlgorithm A machine learning algorithm.
+ * @tparam Metric A metric to assess the quality of a trained model.
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions (should be passed when the
+ *     predictions type is a template parameter in Train methods of
+ *     MLAlgorithm).
+ * @tparam WeightsType The type of weights (should be passed when weighted
+ *     learning is supported, and the weights type is a template parameter in
+ *     Train methods of MLAlgorithm).
+ */
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType = arma::mat,
+         typename PredictionsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType>::PredictionsType,
+         typename WeightsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType,
+                 PredictionsType>::WeightsType>
+class KFoldCV :
+    private CVBase<MLAlgorithm, MatType, PredictionsType, WeightsType>
+{
+ public:
+  /**
+   * Construct an object for running k-fold cross-validation.
+   *
+   * @param k Number of folds (should be at least 2).
+   * @param args Basic constructor arguments for MLAlgortithm (see the CVBase
+   *     constructors for reference).
+   */
+  template<typename... CVBaseArgs>
+  KFoldCV(const size_t k, const CVBaseArgs&... args);
+
+  /**
+   * Run k-fold cross-validation.
+   *
+   * @param args Arguments for MLAlgorithm (in addition to the passed
+   *     ones in the constructor).
+   */
+  template<typename... MLAlgorithmArgs>
+  double Evaluate(const MLAlgorithmArgs& ...args);
+
+  //! Access and modify a model from the last run of k-fold cross-validation.
+  MLAlgorithm& Model();
+
+ private:
+  //! A short alias for CVBase.
+  using Base = CVBase<MLAlgorithm, MatType, PredictionsType, WeightsType>;
+
+  //! The number of bins in the dataset.
+  const size_t k;
+
+  /**
+   * Variables for storing the extended (by repeating the first k - 2 bins)
+   * dataset.
+   */
+  MatType xs;
+  PredictionsType ys;
+  WeightsType weights;
```

Comment:
Can you put one comment per line here?  Otherwise Doxygen will pick up that
entire comment for just the variable xs, and the documentation will be empty for
both ys and weights.